memote/suite/collect.py

```diff
@@ -109,6 +110,56 @@ def _collect_git_info(self):
                 " ")
             self._meta["commit_hash"] = self.commit.hexsha
 
+    def pytest_runtest_protocol(self, item):
+        """
+        Collect test function docstring for tooltip text in the reports.
+
+        Parameters
+        ----------
+        item : pytest.Item
+            A test item objects, i.e. a test function found by pytest.
+
+        """
+        module_name = item.obj.__module__
+        item_name = str(item.name).lower()
+
+        if module_name not in self._data:
+            self._data[module_name] = dict()
+
+        if item_name not in self._data[module_name]:
+            self._data[module_name][item_name] = dict()
+
+        if item.obj.__doc__:
+            docstring = " ".join(FunctionDoc(item.obj)["Summary"])
+            self._data[module_name][item_name]["summary"] = docstring
+        else:
+            self._data[module_name][item_name]["summary"] = "No summary " \
+                                                            "available."
+
```

Comment:
This is a perfect use-case for `dict.setdefault` btw. I would rewrite to the following:
```python
module = self._data.setdefault(item.obj.__module__, dict())
case = module.setdefault(str(item.name).lower(), dict())
case["summary"] = docstring
```